_episodes/07-web-app.md

```diff
@@ -0,0 +1,234 @@
+
+If we call the `plot` function (method) directly on the `pandas` dataframe, we
+get a nice default style:
+
+~~~
+table.plot();
+~~~
+{: .language-python}
+
+![Default Dataframe Plot](../fig/07_default_df_plot.png)
+
+Indeed, we get a legend and an `xlabel`... for free! Here, we can see the
+value of working with *labelled* data.
+
+If we wanted to plot the position of a single runner, say, Jane, we would do
+(see previous episode):
+
+~~~
+table['Jane'].plot();
+~~~
+{: .language-python}
+
+![Default Timeseries Plot](../fig/07_default_ts_plot.png)
+
+If we wanted to plot the positions only up to a certain time point, say, up to
+the third time point, we would do:
+
+~~~
+table[:3].plot();
+~~~
+{: .language-python}
+
+![Plot for a Subset of Rows](../fig/07_subset_df_plot.png)
+
+## Selecting and Plotting them Interactively
+
+[`ipywidgets`](https://ipywidgets.readthedocs.io/) is a library for exploring
+code and data interactively. We import it as follows:
+
+~~~
+import ipywidgets as widgets
+~~~
+{: .language-python}
+
+It provides a function, named `interact`, which allows us to create basic user
+interface (UI) controls. For example, we may use a dropdown menu for
+selecting the runner (Emma or Jane) whose position we want to visualize.
+Likewise, we may use a slider for selecting the end value of the time points.
+
+To begin with, we need to write a function where the parameter is the variable
+we want to control (interact with). In the first case, this could be simply:
+
+~~~
+def f(x):
+    table[x].plot();
+~~~
+{: .language-python}
+
+Then, we need to call `interact` with this function as the first argument and
+the variable to control as the second argument:
+
+~~~
+widgets.interact(f, x={'Emma', 'Jane'});
+~~~
+{: .language-python}
+
+![Screenshot of Interactive Plot with Dropdown](../fig/07_dropdown_screenshot.png)
+
+> ## Choosing Descriptive Function and Variable Names
+>
+> 1. Write a function to plot the timeseries up to a given time point.
+> 2. What do you think about the names of your function and its parameter?
+>
+> > ## Solution
+> >
+> > 1. To plot the timeseries (which are the columns of dataframe `table`) up
+> > to time point `xx`, we can use the following function:
+> > ~~~
+> > def ff(xx):
+> >     table[:xx].plot();
+> > ~~~
+> > {: .language-python}
+> >
+> > Here we could have a discussion about whether `xx` is included or not: If
+> > `xx` is a row index, it is excluded (as we have seen in episode 02 on
+> > lists). If `xx` is an actual time value (timestamp), then it is included.
+> >
+> > 2. We have named this function `ff` because we already had a function
+> > named `f` (this habit comes from a sub-culture in the R community). To
+> > make the code easier to understand and maintain, it would be better to
+> > name it, say, `truncate_timeseries`; likewise, the `xx` parameter could be
+> > renamed into, say, `end_time`.
+> {: .solution}
+{: .challenge}
+
+To create a slider, we pass the variable an integer:
```

Comment:
something grammatically missing?

Reply:
Nope. It's like when you say, for example, "I'm assigning a variable a given value" or, from the [source episode](https://github.com/swcarpentry/python-novice-inflammation/blob/eb04b9a/_episodes/01-intro.md), "we can assign each person a unique identifier."

Reply:
ok for the assigning (considering your example you would not say: "I'm passing a variable a given value" or "we can pass each person a unique identifier."  but rather add "to"), for passing it sounds awkward to me. However, as English is not my first language I won't argue, though consider that the students may also get confused

Reply:
You may say either "assigning variable A value B" or "assigning value B to variable A" (kind of like "I'm telling you a story" vs. "I'm telling a story to you" -- it means the same thing from a purely logical perspective, but the connotation/nuance is different). When speaking I will say both at some point, and when I explain I'll naturally say/add "B to A" because it makes it very explicit.